cases/src/org/commcare/cases/util/CasePurgeFilter.java

```diff
@@ -119,21 +123,22 @@ private void setIdsToRemoveWithNewExtensions(IStorageUtilityIndexed<Case> caseSt
                 nodeStatus |= STATUS_RELEVANT;
             }
 
-            graph.addNode(c.getCaseId(), new int[]{nodeStatus, c.getID()});
+            internalCaseDAG.addNode(c.getCaseId(), new int[]{nodeStatus, c.getID()});
 
             for (CaseIndex index : indexHolder) {
-                graph.setEdge(c.getCaseId(), index.getTarget(), index.getRelationship());
+                internalCaseDAG.setEdge(c.getCaseId(), index.getTarget(), index.getRelationship());
             }
             indexHolder.removeAllElements();
         }
 
-        propogateRelevance(graph);
-        propogateAvailabile(graph);
-        propogateLive(graph);
+        invalidEdgesWereRemoved = removeInvalidEdges();
```

Comment:
The fact that this directly modifies the DAG doesn't really feel right to me. Is it super hard to feed it down the line and  have it return back a list of the cases to be removed? you could pass in the "toRemove" vector as an object and have the method populate it, which would let you retain the scalar boolean return code.